@shared/api/internal.ts

```diff
@@ -313,20 +319,20 @@ export const getAccountBalances = async ({
 }): Promise<AccountBalancesInterface> => {
   const { networkUrl, networkPassphrase } = networkDetails;
 
-  const dataProvider = new DataProvider({
-    serverUrl: networkUrl,
-    accountOrKey: publicKey,
-    networkPassphrase,
-    metadata: {
-      allowHttp: networkUrl.startsWith("http://"),
-    },
-  });
-
   let balances: any = null;
   let isFunded = null;
   let subentryCount = 0;
 
   try {
+    const dataProvider = new DataProvider({
```

Comment:
I found this edge case while testing coincidentally, if DataProvider throws when we instantiate it then we hit an uncaught error, so moved it into the existing try/catch.